plenum/common/ledger_manager.py

```diff
@@ -841,27 +845,54 @@ def getCatchupReqs(self, consProof: ConsistencyProof):
         # sufficient ConsProof in `preCatchupClbk`
         start = getattr(consProof, f.SEQ_NO_START.nm)
         end = getattr(consProof, f.SEQ_NO_END.nm)
-        batchLength = math.ceil((end-start)/nodeCount)
+        ledger_id = getattr(consProof, f.LEDGER_ID.nm)
+        return self._generate_catchup_reqs(start, end, ledger_id, node_count)
+
+    @staticmethod
+    def _generate_catchup_reqs(start, end, ledger_id, node_count):
+        batch_length = math.ceil((end - start) / node_count)
         reqs = []
         s = start + 1
-        e = min(s + batchLength - 1, end)
-        for i in range(nodeCount):
-            req = CatchupReq(getattr(consProof, f.LEDGER_ID.nm), s, e, end)
+        e = min(s + batch_length - 1, end)
+        for i in range(node_count):
+            req = CatchupReq(ledger_id, s, e, end)
             reqs.append(req)
             s = e + 1
-            e = min(s + batchLength - 1, end)
+            e = min(s + batch_length - 1, end)
             if s > end:
                 break
         return reqs
 
+    @staticmethod
+    def _get_merged_catchup_txns(existing_txns, new_txns):
+        """
+        Merge any newly received txns during catchup with already received txns
+        :param existing_txns:
+        :param new_txns:
+        :return:
+        """
+        idx_to_remove = []
+        for i, (seq_no, _) in enumerate(existing_txns):
```

Comment:
Why do we call `enumerate` here?
Are `existing_txns` and `new_txns` sorted by key (seqNo)?